agent/containers/images/image-test-script.sh

```diff
@@ -0,0 +1,333 @@
+#!/bin/bash
+
+# A demonstration of how to use pbench-tool-meister-start with an existing
+# Redis server and one or more remote hosts
+export CONTROLLER=$(hostname -f)
+
+DISTRO_DEFAULT="centos-8"
+TAG_DEFAULT="a933ae45e"
+REDIS_HOST_DEFAULT=$CONTROLLER
+REDIS_PORT_DEFAULT=17001
+PBENCH_RUN_DEFAULT=/var/tmp/test-run-dir
+
+while :; do
+    read -p "Specify distribution to test (or press 'ENTER' for default ${DISTRO_DEFAULT}): "
+    DISTRO=${REPLY:-${DISTRO_DEFAULT}}
+    read -p "Specify image tag to test (or press 'ENTER' for default ${TAG_DEFAULT}): "
+    TAG=${REPLY:-${TAG_DEFAULT}}
+
+    printf -- "Pulling tool-data-sink image to test...\n"
+    if podman pull quay.io/pbench/pbench-agent-tool-data-sink-$DISTRO:$TAG ; then
+        printf -- "Pull complete!\n\n"
+        break
+    else
+        printf -- "Distro/tag combination could not be found, please try again.\n\n"
+    fi
+done
+
+while :; do
+    read -p "Specify redis host (or press 'ENTER' for default '$REDIS_HOST_DEFAULT'): "
+    REDIS_HOST=${REPLY:-${REDIS_HOST_DEFAULT}}
+    if ping -c 1 -W 1 ${REDIS_HOST} 2>&1 1>/dev/null ; then
+        printf -- "Host pinged and validated!\n\n"
+        break
+    else
+        printf -- "Host specified either does not exist or is unreachable, please try again:\n"
+    fi
+done
+
+while :; do
+    read -p "Specify Redis port (or press 'ENTER' for default port '${REDIS_PORT_DEFAULT}'): "
+    let REDIS_PORT=${REPLY:-${REDIS_PORT_DEFAULT}}
+    if ((1 <= $REDIS_PORT <= 65535)); then
+        break
+    else
+        printf -- "Port must be a non-zero numeric value; please retry\n"
+    fi
+done
+
+export REDIS_HOST REDIS_PORT
+export PBENCH_REDIS_SERVER="${REDIS_HOST}:${REDIS_PORT}"
+
+function wait_keypress {
+    echo "Press any key to continue"
+    while :; do
+        read -t ${1} -n 1
+        if [[ ${?} = 0 ]]; then
+            return 0
+        else
+            echo "Waiting for a keypress"
+        fi
+    done
+}
+
+clear
+
+printf -- "We are now ready to begin the guided pbench image test process. \n\n\n"
+
+wait_keypress 120
+
+# The pbench_run directory is the path to the directory which is used as the
+# persistent volume for the Tool Data Sink container, where the configuration
+# as well as the collected data will be stored.
+
+export pbench_run=$PBENCH_RUN_DEFAULT
+while [[ -d $pbench_run ]]; do
```

Comment:
Hmm; if they specify an alternate `pbench_run` to your `read` which isn't a valid directory name, you'll exit the loop but then fail later trying to create it.